base/src/com/google/idea/blaze/base/qsync/ProjectQuerierImpl.java

```diff
@@ -120,15 +121,17 @@ public PostQuerySyncData update(
       ProjectDefinition currentProjectDef, PostQuerySyncData previousState, BlazeContext context)
       throws IOException, BuildException {
 
-    Optional<VcsState> vcsState = getVcsState(context);
-    SyncQueryStatsScope.fromContext(context)
-        .ifPresent(stats -> stats.setSyncMode(SyncQueryStats.SyncMode.DELTA));
-    logger.info(
-        String.format(
-            "Starting partial query update; upstream rev=%s; snapshot path=%s",
-            vcsState.map(s -> s.upstreamRevision).orElse("<unknown>"),
-            vcsState.flatMap(s -> s.workspaceSnapshotPath).map(Object::toString).orElse("<none>")));
-
+    Optional<VcsState> vcsState = Optional.empty();
+    if (BlazeUserSettings.getInstance().getExpandSyncToWorkingSet()) {
```

Comment:
Probably other option should be added for QS yet even QS-relevant delta calculation will be taxng in some cases.

Also Sync to Working Set is enabled by default, iirc. So default QS behavior won't change.